cscs-checks/system/basic/file_system_check.py

```diff
@@ -0,0 +1,174 @@
+            },
+            'dom:login': {
+                'real_time': (0.1, None, 0.1, 's')
+            }
+        }
+        self.executable = 'time ls'
+        self.executable_opts = [variant]
+
+
+@rfm.parameterized_test(['/project/csstaff/bignamic'],
+                        ['/users/bignamic'],
+                        ['/scratch/snx1*/bignamic'],
+                        ['/scratch/snx3*/bignamic'])
+class FileSystemDuDirCheck(FileSystemCommandCheck):
+    def __init__(self, variant):
+        super().__init__()
+        self.descr = 'du of directory'
+
+        # TODO: is it possible to append a pattern?
+        self.perf_patterns = {
+            'real_time': sn.extractsingle(r'\nreal.+m(?P<real_time>\S+)s',
+                                          self.stderr, 'real_time', float),
+            # TODO: this should be solved with parametrized user
+            'size': sn.extractsingle(r'(?P<size>\S+).+/bignamic',
+                                     self.stdout, 'size', float)
+        }
+
+        # TODO: system is not always relevant
+        self.reference = {
+            '/project/csstaff/bignamic': {
+                'size': (1000, None, 0.1, 'MB'),
+                'real_time': (5.0, None, 0.1, 's')
+            },
+            '/users/bignamic': {
+                'size': (900, None, 0.1, 'MB'),
+                'real_time': (5.0, None, 0.1, 's')
+            },
+            '/scratch/snx3*/bignamic': {
+                'size': (900, None, 0.1, 'MB'),
+                'real_time': (5.0, None, 0.1, 's')
+            }
+        }
+        self.executable = 'time du -mhs --block-size=1M'
+        self.executable_opts = [variant]
+
+
+# TODO: can we parametrize the user?
+# TODO: avoid wildcard in folder name
+@rfm.parameterized_test(['/scratch/snx1*/bignamic'],
+                        ['/scratch/snx3000/bignamic'])
+class FileSystemTouchFileCheck(FileSystemCommandCheck):
+    def __init__(self, variant):
+        super().__init__()
+        self.descr = 'touch of file'
+        self.executable = 'time touch'
+        self.test_file = variant + '/reframe_touch_test_file'
+        self.executable_opts = [self.test_file]
+
+    @rfm.run_after('run')
+    def delete_test_file(self):
+        if os.path.exists(self.test_file):
+            os.remove(self.test_file)
+
+
+@rfm.parameterized_test(['/apps/daint/system/etc/BatchDisabled'],
+                        ['/etc/opt/slurm/cgroup.conf'],
+                        ['/etc/opt/slurm/plugstack.conf'],
+                        ['/etc/opt/slurm/slurm.conf'],
+                        ['/etc/opt/slurm/topology.conf'],
+                        ['/etc/opt/slurm/node_prolog.sh'],
+                        ['/etc/opt/slurm/node_epilog.sh'],
+                        ['/etc/opt/slurm/gres.conf'])
+class FileSystemCatCheck(FileSystemCommandCheck):
+    # TODO: find correct test name
+    def __init__(self, variant):
+        super().__init__()
+        self.descr = 'cat of file'
```

Comment:
If we have good names, perhaps such descriptions are redundant.

Reply:
Done